pkg/chunk/purger/purger.go

```diff
@@ -0,0 +1,576 @@
+package purger
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/weaveworks/common/user"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/purger/purgeplan"
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+const millisecondPerDay = int64(24 * time.Hour / time.Millisecond)
+
+var separatorString = string([]byte{model.SeparatorByte})
+
+type deleteRequestWithLogger struct {
+	chunk.DeleteRequest
+	logger log.Logger
+}
+
+// DataPurgerConfig holds config for DataPurger
+type DataPurgerConfig struct {
+	EnablePurger    bool   `yaml:"enable_purger"`
+	NumWorkers      int    `yaml:"num_workers"`
```

Comment:
I like `NumWorkers` because it makes it clear that user has to define a number while `Worker` word doesn't tell whether its a complex object to provide definition of workers or a simple integer type. What do you think?

Reply:
I think `Workers int` is quite clear ("number of workers"), but `NumWorkers` is fine too.